xcodebuild/utils.go

```diff
@@ -0,0 +1,383 @@
+package xcodebuild
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/progress"
+	cache "github.com/bitrise-io/go-xcode/xcodecache"
+	"github.com/kballard/go-shellquote"
+)
+
+// XcprettyTool ...
+const XcprettyTool = "xcpretty"
+
+const (
+	none           = "none"
+	untilFailure   = "until_failure"
+	retryOnFailure = "retry_on_failure"
+)
+
+var xcodeCommandEnvs = []string{"NSUnbufferedIO=YES"}
+
+var testRunnerErrorPatterns = []string{
+	timeOutMessageIPhoneSimulator,
+	timeOutMessageUITest,
+	earlyUnexpectedExit,
+	failureAttemptingToLaunch,
+	failedToBackgroundTestRunner,
+	appStateIsStillNotRunning,
+	appAccessibilityIsNotLoaded,
+	testRunnerFailedToInitializeForUITesting,
+	timedOutRegisteringForTestingEvent,
+	testRunnerNeverBeganExecuting,
+	failedToOpenTestRunner,
+}
+
+// On performance limited OS X hosts (ex: VMs) the iPhone/iOS Simulator might time out
+//  while booting. So far it seems that a simple retry solves these issues.
+const (
```

Comment:
Let's move this next to the other `const` declarations. Also, should we merge them?

Reply:
Exported consts moved to xcodebuild.go, only these private consts remained here.